data/data-pipeline/data_pipeline/etl/score/etl_utils.py

```diff
@@ -305,3 +312,89 @@ def create_codebook(
     return merged_codebook_df[constants.CODEBOOK_COLUMNS].rename(
         columns={constants.CEJST_SCORE_COLUMN_NAME: "Description"}
     )
+
+
+# pylint: disable=too-many-arguments
+def compare_to_list_of_expected_state_fips_codes(
+    actual_state_fips_codes: typing.List[str],
+    nation_expected: bool = True,
+    puerto_rico_expected: bool = True,
+    island_areas_expected: bool = True,
+    additional_fips_codes_not_expected: typing.List[str] = None,
+    dataset_name: str = None,
+) -> None:
+    """Check whether a list of state/territory FIPS codes match expectations.
+
+    Args:
+        actual_state_fips_codes (List of str): Actual state codes observed in data
+        nation_expected (bool, optional): Do you expect the nation (DC & states) to be
+            represented in data?
+        puerto_rico_expected (bool, optional): Do you expect PR to be represented in data?
+        island_areas_expected (bool, optional): Do you expect Island Areas to be represented in
+            data?
+        additional_fips_codes_not_expected (List of str, optional): Additional state codes
+            not expected in the data. For example, the data may be known to be missing
+            data from Maine and Wisconsin.
+        dataset_name (str, optional): The name of the data set, used only in printing an
+            error message. (This is helpful for debugging during parallel etl runs.)
+
+    Returns:
+        None: Does not return any values.
+
+    Raises:
+        ValueError: if lists do not match expectations.
+    """
+    # Setting default argument of [] here to avoid mutability problems.
+    if additional_fips_codes_not_expected is None:
+        additional_fips_codes_not_expected = []
+
+    # Cast input to a set.
+    actual_state_fips_codes_set = set(actual_state_fips_codes)
+
+    # Start with the list of all FIPS codes for all states and territories.
+    expected_states_set = set(get_state_fips_codes(settings.DATA_PATH))
+
+    # If nation (states and DC) are not expected to be included, remove it from the
+    # expected
+    # states set.
+    if not nation_expected:
+        expected_states_set = expected_states_set - set(TILES_NATION_FIPS_CODE)
+
+    # If Puerto Rico is not expected to be included, remove it from the expected
+    # states set.
+    if not puerto_rico_expected:
+        expected_states_set = expected_states_set - set(
+            TILES_PUERTO_RICO_FIPS_CODE
+        )
+
+    # If island areas are not expected to be included, remove them from the expected
+    # states set.
+    if not island_areas_expected:
+        expected_states_set = expected_states_set - set(
+            TILES_ISLAND_AREA_FIPS_CODES
+        )
+
+    # If additional FIPS codes are not expected to be included, remove them from the
```

Comment:
this is a really annoying quibble and i am sorry. 

Above you have a control statement to fill additional fips codes if it is none, and then to  do this. I wonder why not make it default as None of type set, and then "if not none" do this (vs fill it with the empty list if None and do this calculation every time)? This is a nitty nit but 🤷 i learn from the best

Reply:
Hmm, interesting. I see your point. I started without having the nation explicitly defined so it made sense to subtract, but now we have everything. 

The one weird part of starting "positively" would be you'd have to say "ALASKA_AND_HAWAII_INCLUDED=TRUE" for some sources but then ALSO that "15" is an expected missing state. (Some have Alaska but not Hawaii, etc.) 

I'm tempted to just leave it for now to avoid the refactor?

Reply:
that's fine by me